packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx

```diff
@@ -66,32 +66,23 @@ export const ChangePasswordDialogProvider: FC<
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [isSubmitDisabled, setIsSubmitDisabled] = useState<boolean>(true);
 
-  const validateNewPassword = () => {
-    if (newPassword.length > 0 && confirmNewPassword.length === 0) {
-      const validation = validatePassword(
-        { password: newPassword, confirm: newPassword },
-        lang,
-      );
-      if (!validation.success) {
-        setError(validation.error.errors[0].message);
-        return;
-      }
+  useEffect(() => {
+    if (oldPassword === newPassword || oldPassword === confirmNewPassword) {
+      setError(lang.strings.lockscreen.sameOldAndNewPassword);
+      return;
     }
 
-    if (newPassword.length === 0 && confirmNewPassword.length > 0) {
-      const validation = validatePassword(
-        { password: confirmNewPassword, confirm: confirmNewPassword },
-        lang,
-      );
-      if (!validation.success) {
-        setError(validation.error.errors[0].message);
-        return;
-      }
-    }
+    let _password: string | null = null;
+    if (newPassword.length > 0) _password = newPassword;
+    if (confirmNewPassword.length > 0) _password = confirmNewPassword;
+
+    let _confirm: string | null = null;
+    if (confirmNewPassword.length > 0) _confirm = confirmNewPassword;
+    if (newPassword.length > 0) _confirm = newPassword;
```

Comment:
```suggestion
    const _password = newPassword || confirmNewPassword || null;
    const _confirm = confirmNewPassword || newPassword || null;
```

Reply:
Empty string is `false` is JS. Try out the suggested code.

Reply:
ohh sorry! I got confused by `??` operator

Reply:
Fixed